### PR TITLE
show system forms when searching for "Unknown Users" in Submit History Reports

### PR DESCRIPTION
**Pull request by jmtroth0**
adds system forms to "Unknown User" section when searching, which is my understanding of what this ticket is looking for https://manage.dimagi.com/default.asp?285161

The other option was to add a search by `xmlns` instead of user_id, but that seems more fragmented than the system user_id, as I was finding at least 3 different xmlns's that we use to denote a system transaction.

@esoergel buddy: @proteusvacuum

### CODE REVIEW

**Comment by jmtroth0** [comment]
first commit is the relevant change

File: corehq/apps/es/users.py [hunk 1]
```diff
@@ -98,6 +98,12 @@ def unknown_users():
     """
     return filters.doc_type("UnknownUser")
 
+def system_users():
+    """
+    Returns system users. system users have a user_id of system
+    """
+    return username("system")
```
Comment (proteusvacuum) [inline corehq/apps/es/users.py]: might be good to use the constant for this: https://github.com/dimagi/commcare-hq/blob/9aa3a8ea8cd4bfd4782eb59b01b4cbb9b86c1e4a/corehq/apps/users/util.py#L22

Comment (jmtroth0) [reply corehq/apps/es/users.py]: OOOOOH, i hadn't seen that. I'll also update the demo user one

File: corehq/apps/reports/filters/users.py [hunk 2]
```diff
@@ -360,6 +360,7 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs, request_user):
             user_type_filters.append(user_es.admin_users())
         if HQUserType.UNKNOWN in user_types:
             user_type_filters.append(user_es.unknown_users())
+            user_type_filters.append(user_es.system_users())
```
Comment (proteusvacuum) [inline corehq/apps/reports/filters/users.py]: I wonder if we want to expose this as a separate option too, it seems like most people know they are looking for `system` when looking for HQ submitted forms

File: corehq/apps/es/users.py (hunk 1 above)
Comment (jmtroth0) [reply corehq/apps/es/users.py]: Although it's possible this module is trying to not reach into other ones too much

Comment (proteusvacuum) [reply corehq/apps/es/users.py]: ^ if that's true then I would say having the string "system" is an invisible coupling which is harder to reason / know about. 
You could bump this out to `filters/users.py` instead which already imports `util` and do 

```py
user_type_filters.append(user_es.username(util.SYSTEM_USER_ID))
```

instead

**Review by esoergel — APPROVED** [review]
This looks good, were you able to validate it locally?

File: corehq/apps/reports/filters/users.py (hunk 2 above)
Comment (esoergel) [reply corehq/apps/reports/filters/users.py]: That'd be good to raise with @djmore9 - he's been looking at this filter a bit lately

**Comment by jmtroth0** [comment]
I wasn't sure how to make a form available to the report, but it runs at least. It should be something like `./manage.py ptop_reindexer_v2 sql-form`?

**Comment by proteusvacuum** [comment]
@jmtroth0 you might have to add `--cleanup` to that command, otherwise I think there is some checkpointing issues that stop new forms from moving to ES locally.

**Pull request merged**